pymc/distributions/distribution.py

```diff
@@ -277,6 +314,23 @@ def __new__(
                 "for a standalone distribution."
             )
 
+        if "name" in kwargs:
+            name = kwargs.pop("name")
+        elif len(args) > 0 and isinstance(args[0], string_types):
+            name = args[0]
+            args = args[1:]
+        else:
+            name = _extract_target_of_assignment(2)
+            if name is None:
+                raise TypeError(
+                    "Name could not be inferred for variable from surrounding "
+                    "context. Pass a name explicitly as the first argument to "
+                    "the Distribution."
+                )
+
+        if not isinstance(name, string_types):
+            raise TypeError(f"Name needs to be a string but got: {name}")
+
```

Comment:
```suggestion
        name = kwargs.pop("name", None)
        if name is None:
            if args and isinstance(args[0], string_types):
                # Name was provided as the first argument
                name = args[0]
                args = args[1:]
            else:
                # Try to infer name from the outer context.
                # This may raise an error, but then there's nothing else we can do.
                name = _extract_target_of_assignment(2)

        if not isinstance(name, string_types):
            raise TypeError(f"Name needs to be a string but got: {name}")

```